### PR TITLE
Use paradox with its default Compile configuration

### PR DESCRIPTION
**Pull request by 2m**
*This is a second attempt to fix the underlying cause for #141 and supersedes #141.*

Originally sbt-site would introduce a new `Paradox` configuration, and use it to scope all sbt-paradox settings. However that means that when adding sbt-site to a project, that already uses sbt-paradox, this configuration would have to be added to all of the paradox settings.

Also it would require sbt-site to override some of the sbt-paradox behavior that is has for custom configuration.

This PR changes sbt-site to use the default (`Compile`) scope for integration with sbt-paradox. It keeps the `Paradox` configuration, but only for the sbt-site specific `siteSubdirName` setting.

### CODE REVIEW

File: src/sbt-test/site/can-run-generator-twice/build.sbt [hunk 1]
```diff
@@ -6,17 +6,18 @@ val Site1 = config("site1")
 val Site2 = config("site2")
 
 // Apply the default Paradox settings to the `Site1` config
+ParadoxPlugin.paradoxSettings(Site1)
 ParadoxSitePlugin.paradoxSettings(Site1)
```
Comment (2m) [inline src/sbt-test/site/can-run-generator-twice/build.sbt]: This is to keep the same usage as one would do when using sbt-site with multiple configurations.

**Comment by ennru** [comment]
This change needs to be reflected in the docs, as well: https://www.scala-sbt.org/sbt-site/generators/paradox.html

**Comment by 2m** [comment]
Yup, thanks! It also needs a bit more tweaking so all of the scripted tests pass.

**Comment by 2m** [comment]
Failures in tests were partly due to a bug in Paradox: https://github.com/lightbend/paradox/pull/312

File: src/main/scala/com/typesafe/sbt/site/util/SiteHelpers.scala [hunk 2]
```diff
@@ -44,7 +44,10 @@ object SiteHelpers {
       ))
 
   def watchSettings(config: Configuration): Seq[Setting[_]] =
-    Compat.watchSettings(config)
+    Compat.watchSettings(ThisScope.in(config))
```
Comment (2m) [inline src/main/scala/com/typesafe/sbt/site/util/SiteHelpers.scala]: I hope `config: Configuration == ThisScope.in(config)`

File: src/main/scala/com/typesafe/sbt/site/Compat.scala [hunk 3]
```diff
@@ -49,12 +49,12 @@ object Compat {
...
-  def watchSettings(config: Configuration): Seq[Setting[_]] =
+  def watchSettings(scope: Scope): Seq[Setting[_]] =
     Seq(
       ConfigGlobal / watchSources += new Source(
-        base = (sourceDirectory in config).value,
-        includeFilter = (includeFilter in config).value,
-        excludeFilter = (excludeFilter in config).value
+        base = (sourceDirectory in scope).value,
+        includeFilter = (includeFilter in scope).value,
+        excludeFilter = (excludeFilter in scope).value
```
Comment (2m) [inline src/main/scala/com/typesafe/sbt/site/Compat.scala]: I made it more general, so I could add watchsettings not only under Configuration scope, but also under a task scope.

**Comment by 2m** [comment]
@dwijnand I would really appreciate a review here on the changes around `watchSources` as I am not really familiar with that sbt area.

I tested it locally and it picked up the changes when running `previewAuto`. So at least we know it works.

File: src/main/scala/com/typesafe/sbt/site/util/SiteHelpers.scala (hunk 2 above)
Comment (dwijnand) [reply src/main/scala/com/typesafe/sbt/site/util/SiteHelpers.scala]: Yes, `key in config == key in ThisScope.in(config)`

**Pull request merged**